src/System/Hapistrano.hs

```diff
@@ -225,22 +255,29 @@ deployedReleases
   -> Hapistrano [Release]
 deployedReleases deployPath = do
   let rpath = releasesPath deployPath
-  xs <- exec (Find 1 rpath :: Find Dir)
+  xs <- exec (Find 1 rpath :: Find Dir) Nothing
   ps <- stripDirs rpath (filter (/= rpath) xs)
   (return . sortOn Down . mapMaybe parseRelease)
     (dropWhileEnd (== '/') . fromRelDir <$> ps)
 
 -- | Return a list of successfully completed releases sorted newest first.
 
-completedReleases
-  :: Path Abs Dir      -- ^ Deploy path
+releasesWithState
+  :: DeployState       -- ^ Selector for failed or successful releases
+  -> Path Abs Dir      -- ^ Deploy path
   -> Hapistrano [Release]
-completedReleases deployPath = do
-  let cpath = ctokensPath deployPath
-  xs <- exec (Find 1 cpath :: Find File)
-  ps <- stripDirs cpath xs
-  (return . sortOn Down . mapMaybe parseRelease)
-    (dropWhileEnd (== '/') . fromRelFile <$> ps)
+releasesWithState selectedState deployPath = do
+  releases <- deployedReleases deployPath
+  filterM (
+    fmap ((\bool -> if selectedState == Success then bool else not bool) . stateToBool)
+     . deployState deployPath Nothing
+    ) releases
+  where
+    stateToBool :: Maybe DeployState -> Bool
+    stateToBool mDeployState =
+      case mDeployState of
+        (Just Fail) -> False
+        _ -> True
```

Comment:
We can make this shorter this way:
```suggestion
    stateToBool :: Maybe DeployState -> Bool
    stateToBool (Just Fail) = False
    stateToBool _ = True
```

Reply:
I feel a bit embarrased about this one 😅 I'll be changing this, thanks.